sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1238,6 +1238,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val SORT_IN_MEM_FOR_LIMIT_THRESHOLD =
+    buildConf("spark.sql.limit.sortInMemThreshold")
+      .internal()
+      .doc("In sql like 'select x from t order by y limit m', if m is under this threshold, " +
+          "sort in memory, otherwise do a global sort with disk.")
+      .intConf
+      .createWithDefault(2000)
```

Comment:
A numeric limit threshold seems to hard to determine, how about a boolean config (default true) we can enable `TakeOrderedAndProjectExec` or not? cc @cloud-fan

Reply:
what if users only have a few queries which have large limit and they want to disable the top n sort? I feel this config is more flexible than a boolean flag.

Reply:
Oh, yeah, reasonable.

Reply:
Isn't 2000 too small for this?

Reply:
I would suggest `Int.Max` as the default value, which preserves the previous behavior. Users can tune it w.r.t. their workload.

Reply:
Yeah, I agree.